### PR TITLE
Bug 1219678 - Implement number keyboard test from gip to gij, r=timdream

### PR DESCRIPTION
**Pull request by raylin**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=66ac0dc10073bfa300f9cfb568b936bf3a86e89e

File: apps/keyboard/test/marionette/number_keyboard_test.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    });
+
+    test('number input', function() {
+      keyboardTestApp.numberInput.tap();
+
+      systemInputMgmt.waitForKeyboardFrameDisplayed();
+      systemInputMgmt.switchToActiveKeyboardFrame();
+
+      assert.equal(
+        keyboard.getCurrentInputType(), typeGroupMap.number);
```
Comment (timdream) [inline apps/keyboard/test/marionette/number_keyboard_test.js]: Please add another assertion here proving that the current active layout has no alphabet keys.

File: apps/keyboard/test/marionette/number_keyboard_test.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      // Switch to test app frame.
+      client.switchToFrame();
+      client.apps.switchToApp(KeyboardTestApp.ORIGIN);
+
+      assert.equal(
+        inputString, keyboardTestApp.numberInput.getAttribute('value'));
+    });
+  });
+
+  suite('<textarea> tests', function() {
```
Comment (timdream) [inline apps/keyboard/test/marionette/number_keyboard_test.js]: Copy paste fail... this part is duplicate and needs to be removed.

File: apps/keyboard/test/marionette/lib/keyboard.js [hunk 3]
```diff
@@ -97,28 +127,25 @@ Keyboard.prototype = {
...
+  getCurrentInputType: function () {
+    return this.client.executeScript(
+      'return window.wrappedJSObject.app.getBasicInputType();');
+  },
 
-    var pageSwitchingKey = this.getPageSwitchingKey(index);
-    pageSwitchingKey.tap();
+  getCurrentInputMode: function () {
+    return this.client.executeScript(
+      'return window.wrappedJSObject.app.inputContext.inputMode;');
```
Comment (timdream) [inline apps/keyboard/test/marionette/lib/keyboard.js]: `wrappedJSObject`: Probing JS values in the app context can only be used as helper function, not something the tests should assert against.

I am fine if we reimpl these functions from Gip but please make sure assertions are added accordingly (like my previous comment).

File: apps/keyboard/test/marionette/lib/keyboard.js [hunk 4]
```diff
@@ -56,11 +61,36 @@ Keyboard.prototype = {
...
+    this.client.settings.set('keyboard.autocorrect', val);
+  },
+
   getKey: function getKey(key) {
     var keySelector = Keyboard.Selector.key;
 
     if (key >= '0' && key <='9') {
-      this.switchToPage(1);
+      if (this.getCurrentInputType() !== 'number' && 
+          this.getCurrentInputMode() !== 'numeric') {
```
Comment (timdream) [inline apps/keyboard/test/marionette/lib/keyboard.js]: I agree this is a legit use case of helper functions.

File: apps/keyboard/test/marionette/lib/keyboard.js [hunk 5]
```diff
@@ -56,11 +61,36 @@ Keyboard.prototype = {
...
+  },
+
+  get predictiveWord() {
+    return this.client.findElement(Keyboard.Selector.predictiveWord);
+  },
+
+  set autoCorrect(val) {
+    this.client.settings.set('settings-write', true);
+    this.client.settings.set('settings-api-write', true);
+    this.client.settings.set('keyboard.autocorrect', val);
```
Comment (timdream) [inline apps/keyboard/test/marionette/lib/keyboard.js]: Let's remove this API and actually go to the Keyboard Settings panel and toggle these switches (in the tests that uses this API).

**Comment by timdream** [comment]
Please update the commit message to be something more descriptive.

**Pull request closed without merging**